### DISCUSSION TITLE
oauth clientSecret from environmentVariable or file

### DISCUSSION THREAD

**Discussion by chrichrichri** [immich-app/immich#11045] [Q&A] [answered]
I would like to store my immich config file and my `docker-compose` file within a git repository. But I do not want to store the secrets there. Is it currently possible, while I can save all database settings and passwords in a plain-text file, it does not seem to be possible to store `clientSecret` and `clientId` differently than in the config file itself.

Is there any possiblity to laod these from a environment variable or from a file?

**Comment by bo0tzz** [comment]
Immich doesn't have an option for that at the moment.

Personally I use https://github.com/getsops/sops to encrypt secrets and store them in my homelab git repo. I can recommend a setup like that, it's very convenient.

**Comment by chrichrichri** [comment]
Yes, I am already using `age` for this. But it means I need to store the complete config file in there compared to just the secrets, which seems also to not be the optimal solution. But anyhow, thanks for the fast answer. Maybe it can be introduced in upcoming release.

**Reply by bo0tzz** [reply, marked as answer]
You could use something like envsubst to replace placeholders in the config json, or `jq` to merge multiple partial config json files, etc.

**Reply by chrichrichri** [reply]
envsubst was indeed a nice tip. Thanks! I combined it with a service running automatically which should solve the job.

**Comment by bdsoha** [comment]
This is not a solution for _k8s-based_ installs.

**Reply by cterence** [reply]
Here's a [way](https://github.com/cterence/homelab-gitops/commit/f5689d93b7ed211751bf54aeb148be080a682ee8) to solve the problem with Kubernetes. I use an init container to run envsubst on the config file from the readonly configmap. It outputs the result in an emptyDir volume mounted by both init and main containers. The "templated" configmap is the one I provide to immich main container with the IMMICH_CONFIG_FILE env var.

**Reply by phoenix-frozen** [reply]
Thank you, @cterence !!

**Comment by Srylax** [comment]
This would still be very useful when using the NixOs Module

**Comment by timove** [comment]
There has been an update, see [settings](https://search.nixos.org/options?channel=25.11&show=services.immich.settings&query=services.immich.): 

> You can load secret values from a file in this configuration by setting somevalue._secret = "/path/to/file" instead of setting somevalue directly.
